src/app/joke/joke.component.ts

```diff
@@ -23,4 +25,19 @@ export class JokeComponent implements OnInit {
       console.log(this.list);
   });
 }
+randomJoke() {
+  this.router.navigate(['jokes/random']);
+  
 }
+
+jokeDetail() {
+// alert('in joke Details()');
```

Comment:
Why Do we need this method?

Reply:
i forgot to take out that,
previously i thought the requirement is like 
when i click on the Jokelist, i need to geet the joke detail component
but later i unnderstood.
i will change it